server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -166,6 +168,7 @@ protected void doRun() throws Exception {
                 }
                 // We're done, there's no more operations to execute so we resolve the wrapped listener
                 finishRequest();
+                primary.getBulkOperationListener().afterBulk(request.totalSizeInBytes(), System.nanoTime() - startBulkTime);
```

Comment:
I believe this should be moved to before the `finishRequest()`, finishing the request entails running a number of things that happen as part of the listener passed in to this method, and while it could be good to capture the time spent doing those things, right now we do *not* capture them for the replica operation (since they are done after `shardOperationOnReplica` is called.

If we want to stick with tracking only the indexing time (rather than refresh, checkpoint update, etc), we should track the time prior to the `finishRequest()` call

Reply:
I get it, track the time prior to the finishRequest() also ensures that the same events are captured on primary and replic